File: Backend_dot_net_c_sharp_models_and_eums_to_convert_to_dart /Enumeration/RelationTypes.cs
using PeerageHRIS.Resources.Models.Employee.Profiles;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration
{
    public enum RelationTypes
    {
        [Display(Name = "Husband", ResourceType = typeof(EmployeeDependantResource))]
        Husband,
        [Display(Name = "Wife", ResourceType = typeof(EmployeeDependantResource))]
        Wife,
        [Display(Name = "Child", ResourceType = typeof(EmployeeDependantResource))]
        Child,
        [Display(Name = "Father", ResourceType = typeof(EmployeeDependantResource))]
        Father,
        [Display(Name = "Mother", ResourceType = typeof(EmployeeDependantResource))]
        Mother,
        [Display(Name = "Brother", ResourceType = typeof(EmployeeDependantResource))]
        Brother,
        [Display(Name = "Sister", ResourceType = typeof(EmployeeDependantResource))]
        Sister, 
        [Display(Name = "Guardian", ResourceType = typeof(EmployeeDependantResource))]
        Guardian,
        [Display(Name = "Other", ResourceType = typeof(EmployeeDependantResource))]
        Other
    }
}

File: lib/core/enums/relation_types.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'relation_types.g.dart';

@JsonEnum(alwaysCreate: true)
enum RelationTypes {
  @JsonValue(0)
  husband,

  @JsonValue(1)
  wife,

  @JsonValue(2)
  child,

  @JsonValue(3)
  father,

  @JsonValue(4)
  mother,

  @JsonValue(5)
  brother,

  @JsonValue(6)
  sister,

  @JsonValue(7)
  guardian,

  @JsonValue(8)
  other,
}
